src/olympia/scanners/tests/test_views.py

```diff
@@ -275,3 +277,30 @@ def test_get_by_label_and_scanner(self):
             '{}'.format('{}?scanner=wat&label=good'.format(self.url))
         )
         self.assert_json_results(response, expected_results=1)
+
+    def test_get_results_with_blocked_versions(self):
+        self.create_switch('enable-scanner-results-api', active=True)
+        # result labelled as "bad" because its state is TRUE_POSITIVE
+        bad_version = version_factory(addon=addon_factory())
+        ScannerResult.objects.create(
+            scanner=YARA, version=bad_version, state=TRUE_POSITIVE
+        )
+        # result labelled as "bad" because the add-on is blocked.
+        blocked_addon_1 = addon_factory()
+        blocked_version_1 = version_factory(addon=blocked_addon_1)
+        ScannerResult.objects.create(scanner=YARA, version=blocked_version_1)
+        block_1 = Block.objects.create(
+            guid=blocked_addon_1.guid, updated_by=self.user
+        )
+        block_activity_log_save(block_1, change=False)
+        # result labelled as "bad" because the add-on is blocked.
+        blocked_addon_2 = addon_factory()
+        blocked_version_2 = version_factory(addon=blocked_addon_2)
+        ScannerResult.objects.create(scanner=YARA, version=blocked_version_2)
+        block_2 = Block.objects.create(
+            guid=blocked_addon_2.guid, updated_by=self.user
+        )
+        block_activity_log_save(block_2, change=True)
```

Comment:
Maybe add 2 more cases:
- One with a block added *and* edited.
- One with a true positive and a block added

The idea being to guard against possible regression where we'd return the same add-on multiple times.